src/core/constructs/workspace.py

```diff
@@ -171,14 +185,57 @@ def remove_global_instance(cls, caller: "Workspace") -> None:
 
         _GLOBAL_WORKSPACE = None
 
-    def initialize_workspace(self, workspace_configuration: Workspace_Info) -> None:
+    def initialize_workspace(
+        self,
+        settings_info: Settings_Info,
+        backend_info: Backend_Configuration,
+        resource_state_uuid: str,
+        initialization_modules: Optional[List[str]] = [],
+        configuration: Dict = {},
+    ) -> None:
         """Run the configuration needed to initialize a workspace. This should generally only be called by the Core framework itself to ensure that the
         life cycle of a workspace is correctly handled.
 
         Args:
-            workspace_configuration(Workspace_Info)
+            settings_info (Settings_Info): Information to load the `Workspace` settings
+            backend_info (Backend_Configuration): information to connect to the `Backend`
+            resource_state_uuid (str): resource state to execute commands over
+            configuration (Dict): additional configuration
         """
-        raise NotImplementedError
+        try:
+            initialized_backend = load_backend(backend_info)
+            self.set_backend(initialized_backend)
+        except Exception as e:
+            print(f"Could not load the load backend")
+            raise e
+
+        try:
+            top_level_resource_states = (
+                initialized_backend.get_top_level_resource_states()
+            )
+        except Exception as e:
+            raise e
+
+        if not resource_state_uuid in set([x.uuid for x in top_level_resource_states]):
+            raise Exception(
+                f"{resource_state_uuid} not in loaded backend resource states: ({top_level_resource_states})"
+            )
+
+        self.set_resource_state_uuid(resource_state_uuid)
+
+        try:
+            initialized_settings = initialize_settings(settings_info)
+            self.settings = initialized_settings
+        except Exception as e:
+            print(f"Could not load the load backend")
+            raise e
+
+        if initialization_modules:
+            for initialize_module in initialization_modules:
+                try:
+                    module_loader.import_module(initialize_module)
+                except Exception as e:
+                    raise e
```

Comment:
Print something here to help pinpoint the issue, maybe the module name?

Reply:
adding more descriptive exceptions today

Reply:
we should add a message like "Could not initialize the modules"